gp2-stress-experiment.ipynb

```diff
@@ -0,0 +1,933 @@
+    "ds[\"sample_plastic\"] = load_and_scale(folder_name=\"4) R825 3500 um\", scale_factor=pulse_number_sample_plastic)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Adds a coordinate for the spectra and TOF\n",
+    "ds.coords[\"spectrum\"] = sc.Variable([\"spectrum\"], values=np.arange(ds[\"sample\"].shape[1]))\n",
+    "stitched = sc.Dataset(coords={\"tof\": sc.Variable([\"tof\"], values=np.arange(*rebin_parameters, dtype=np.float64))})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def stitch_data(variable_to_stitch):\n",
+    "    print(f\"Stitching: {variable_to_stitch}\")\n",
+    "    stitched[variable_to_stitch] = imaging.stitch(ds[variable_to_stitch],\n",
+    "                                                                           frame_parameters=frame_parameters,\n",
+    "                                                                           frame_shifts=frame_shifts,\n",
+    "                                                                           rebin_parameters=rebin_parameters)\n",
+    "\n",
+    "for var_name in ds.keys():\n",
+    "    stitch_data(var_name)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "tof = stitched.coords['tof']\n",
+    "bin_widths = tof['tof', 1:] - tof['tof', :-1]\n",
+    "\n",
+    "integrated = sc.Dataset()\n",
+    "for dim in stitched.coords.keys():\n",
+    "    integrated.coords[dim] = stitched.coords[dim]\n",
+    "\n",
+    "for k in stitched.keys():\n",
+    "    integrated[k] = sc.sum(stitched[k], \"tof\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "\n",
+    "num_spectra = stitched.coords[\"spectrum\"].shape[0]\n",
+    "bank_width = int(np.sqrt(num_spectra))\n",
+    "integrated = operations.integrate(stitched)\n",
+    "integrated_neighbour_spectra = operations.generate_neighbouring_spectra(num_spectra=num_spectra,\n",
+    "                                                                        bank_width=bank_width)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def mask_non_sample_regions(integrated_dataset, var_name):\n",
+    "    integrated_spectra = integrated_dataset[var_name]\n",
+    "    spectra_masks = [False] * num_spectra\n",
+    "    for i, spec_val in enumerate(integrated_spectra.values):\n",
+    "        if spec_val > masking_threshold:\n",
+    "            spectra_masks[i] =  True\n",
```

Comment:
I'm running into a problem with this:

Trying `spectra_masks = sc.greater(integrated_spectra, sc.Variable(masking_threshold))`
Doesn't quite work since it appears that greater expects a `VariableConstView` whilst we have a `DataArrayView`, is this just a case of some missing bindings?

Reply:
Yes this is missing. Until it is available use `sc.greater(array.data(), var)`.

Reply:
Still running into issues, I've opened https://github.com/scipp/scipp/issues/1178